DOC Add plot stock market example to API docs

alexqiao opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->
#26927 

#### What does this implement/fix? Explain your changes.
Add plot stock market example to API docs

#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
https://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

alexqiao committed:
add plot stock example

ArturoAmorQ reviewed (commented):
Hi @alexqiao thanks for the PR. We normally link the related example to the main paragraph at the beginning of the class, just above the `Read more in the User Guide` phrase.

alexqiao committed:
Update _graph_lasso.py


link the related example to the main paragraph at the beginning of the class, just above the Read more in the User Guide phrase

alexqiao committed:
Merge branch 'main' into branch-plot-stock

alexqiao commented:
I already update the commit, please check whether it is right. Thanks.

ArturoAmorQ commented on sklearn/covariance/_graph_lasso.py:
```diff
@@ -482,9 +485,7 @@ class GraphicalLasso(BaseGraphicalLasso):
-    For a usage example see
-    :ref:`sphx_glr_auto_examples_applications_plot_stock_market.py`
-
+    
```
This extra line still has to be removed, it is causing the linting to fail due to a blank space.

alexqiao replied:
ok, I already delete . Sorry to bother~

alexqiao committed:
Update _graph_lasso.py

delete the blank line

ArturoAmorQ commented on sklearn/covariance/_graph_lasso.py:
```diff
@@ -390,6 +390,9 @@ def __init__(
 
 class GraphicalLasso(BaseGraphicalLasso):
     """Sparse inverse covariance estimation with an l1-penalized estimator.
+    
```
```suggestion

```

ArturoAmorQ committed:
Update sklearn/covariance/_graph_lasso.py

ArturoAmorQ commented on sklearn/covariance/_graph_lasso.py:
```diff
@@ -390,6 +390,8 @@ def __init__(
 
 class GraphicalLasso(BaseGraphicalLasso):
     """Sparse inverse covariance estimation with an l1-penalized estimator.
+    For a usage example see
+    :ref:`sphx_glr_auto_examples_applications_plot_stock_market.py`
```
```suggestion

    For a usage example see
    :ref:`sphx_glr_auto_examples_applications_plot_stock_market.py`.
```

ArturoAmorQ committed:
Update sklearn/covariance/_graph_lasso.py

ArturoAmorQ reviewed (commented):
I think this PR is good to go, thanks @alexqiao. Merging!
